realm/realm-library/src/main/java/io/realm/MutableRealmInteger.java

```diff
@@ -119,45 +120,53 @@ public boolean isValid() {
 
         @Override
         public Long get() {
-            return value;
+            row.checkIfAttached();
+            return (isNull()) ? null : row.getLong(columnIndex);
         }
 
-
-//        // Template code:
-//        if (proxyState.isUnderConstruction()) {
-//            if (!proxyState.getAcceptDefaultValue$realm()) {  // Wat?
-//                return;
-//            }
-//
-//            row.getTable().setLong(columnIndex, row.getIndex(), value, true);
-//            return;
-//        }
-//        row.setLong(columnIndex,value);
         @Override
         public void set(Long value) {
             realm.checkIfValidAndInTransaction();
-            this.value = value;
+
+            if (!proxyState.isUnderConstruction()) {
```

Comment:
I'm not sure.  Adding a task to verify that this all works with the Dynamic API.